### ISSUE TITLE
AWX 19.0.0 is not using the search domains and nameserver specified in the container resolv.conf file

### ISSUE THREAD

**Issue by Dugelon** [ansible/awx#10120]
##### ISSUE TYPE
 - Bug Report

##### SUMMARY
To resolv our internal host listed in our AWX inventories we need to use specific DNS nameservers and search domain.
We add specific "search" domains and "nameserver" in the awx-web and awx-task /etc/resolv.conf file.
This is done by adding a volume and volumeMounts in the awx deployment that use a configmap with resolv.conf content.

With this configuration, the awx-task and awx-web containers can resolv our internal servers hostnames, but the AWX jobs failed to resolv them.

In the previous version of AWX, before 18.0.0, when we install using the docker way we were using the "awx_container_search_domains" parameter available in the "inventory" file.

##### ENVIRONMENT
* AWX version: 19.0.0
* AWX install method: microk8s
* Ansible version:  AWX 19.0.0
* Operating System: Ubuntu 20.04.2 LTS
* Web Browser: Any

##### STEPS TO REPRODUCE

We mount a /etc/resolv.conf file on the containers awx-task and awx-web by editing the awx deployment.
1. Create the configmap using ```kubectl apply -f awx-resolvconf.yml```
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: awx-resolvconf
  namespace: default
data:
  resolv.conf: |
    search default.svc.cluster.local svc.cluster.local cluster.local 1918.lan
    nameserver 10.152.183.10 myinternalnameserver1 myinternalnameserver2
    options ndots:5
```
2. Edit the awx deployment using ```kubecl edit deployment awx``` and add the volume below
```
  volumes:
  - configMap:
      defaultMode: 420
      name: awx-resolvconf
    name: awx-resolvconf
```
3. Edit the awx deployment using ```kubecl edit deployment awx``` and add the VolumeMounts to awx-web and awx-task
```
volumeMounts:
    - mountPath: /etc/resolv.conf
      subPath: resolv.conf
      name: awx-resolvconf
      readOnly: false
```
4. The new containers are created by the deployment, the old are terminating
5. Ping successfully an internal server from the awx-web and awx-task container and check that it add the custom search domain and can resolv the host
``` 
awx-microk8s$ kubectl exec -it awx-769cd4d89f-b5vr4 -c awx-web -- ping smlproxy001
```
![awx-web_resolv_ok](https://user-images.githubusercontent.com/47793633/117254024-33827c80-ae48-11eb-8a13-7b3b80a94b44.png)
```
awx-microk8s$ kubectl exec -it awx-769cd4d89f-b5vr4 -c awx-task -- ping smlproxy001
```
![awx-task_resolv_ok](https://user-images.githubusercontent.com/47793633/117254074-41d09880-ae48-11eb-900d-56648fae6f26.png)

6. Launch a Template that target the same host. AWX failed to resolved the hostname
![job_failed_resolv](https://user-images.githubusercontent.com/47793633/117253888-0a61ec00-ae48-11eb-938d-67fc331ede4d.png)

##### EXPECTED RESULTS

AWX Jobs resolv the hostname using the nameservers and search domain from the container /etc/resolv.conf file

##### ACTUAL RESULTS

AWX Jobs can't resolv the hostname using the nameservers and search domain from the container /etc/resolv.conf file

##### ADDITIONAL INFORMATION

In the previous version of AWX, before 18.0.0, when we install using the docker way we were using the "awx_container_search_domains" parameter available in the "inventory" file to configure the search domains.

 This parameters was taking care to add the search domains in /etc/resolv.conf file of the docker containers. 

Now we need to find a new way to do it.

**Comment by badoqu** [comment]
Have the extact same problem. We used to sepcify the parameter "awx_container_search_domains" in the install inventory file in previous versions. Was working well.

But now we are not able to use search domains for inventory items. 

Only workaround i found is to sepcify the full FQDN in the inventory item names. The problem is that these inventories are populated automatically and use the short domain name...

modifying /etc/resolv.conf in the awx task and web containers did not solve the problem for me too.

**Comment by MEYLSI** [comment]
We face the same problem.
We tried to tweak the cordns config but not working too

**Comment by shanemcd** [comment]
I think the operator is overwriting your changes. Can you try to use these options in the operator? https://github.com/ansible/awx-operator#custom-volume-and-volume-mount-options

**Comment by stanislaspiron** [comment]
in AWX 18.0.0 and above, playbooks are executed in awx-ee container created for each job... so don't try to change awx-task container...

and AWX requires internal k8s coredns to be able to connect to postgre service ... so if you change dns configuration, you can break this communication.

in microk8s, you can change default DNS resolver when enabling dns add-on.

```
microk8s enable dns:10.152.183.10
```

And last question... why do you expect to define dns search domain for automation? isn't it better to define FQDN hostname in playbooks?

**Comment by cisel-dev** [comment]
> 
> 
> I think the operator is overwriting your changes. Can you try to use these options in the operator? https://github.com/ansible/awx-operator#custom-volume-and-volume-mount-options

Thank you for the link, I did not saw this before. We will use it for persistent need.

I didn't know that playbooks were run on awx-ee containers, thanks for this information.

We have already tried to modify the CoreDNS configuration, notably by integrating forwarders to our internal DNS, but this does not solve our problem.

As @stanislaspiron  said, the basic problem is the fact that we don't have a FQDN for hostnames in our inventories. This is caused by the heterogeneity of our environments and that we have many different sources to build them.

So, this is not a bug but more a technical issue. We need to work on the inventory sources to make them usable by AWX without making any changes to the latter(which would be difficult to maintain).

Thanks guy for your time

**Comment by Dugelon** [comment]
> 
> 
> I think the operator is overwriting your changes. Can you try to use these options in the operator? https://github.com/ansible/awx-operator#custom-volume-and-volume-mount-options

Thank you for the link, I did not saw this before. We will use it for persistent need.

I didn't know that playbooks were run on awx-ee containers, thanks for this information.

We have already tried to modify the CoreDNS configuration, notably by integrating forwarders to our internal DNS, but this does not solve our problem.

As @stanislaspiron said, the basic problem is the fact that we don't have a FQDN for hostnames in our inventories. This is caused by the heterogeneity of our environments and that we have many different sources to build them.
 
So, this is not a bug but more a technical issue. We need to work on the inventory sources to make them usable by AWX without making any changes to the latter(which would be difficult to maintain).

Thanks guy for your time
